### PR TITLE
Chat: scrollable integration enhancements

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: packages/devextreme/js/__internal/ui/chat/messagelist.ts [hunk 1]
```diff
@@ -63,47 +58,42 @@ class MessageList extends Widget<Properties> {
...
-      && this._isAttached(target)
-      && isElementVisible(target as HTMLElement)
-    ) {
-      this._scrollContentToLastMessage();
+    if (!isElementInDom($(target)) || !isElementVisible(target as HTMLElement)) {
+      return;
+    }
 
-      this._suppressResizeHandling = false;
+    if (!isDefined(this._containerClientHeight)) {
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/chat/messagelist.ts]:
```suggestion
    const isInitialRendering = !isDefined(this._containerClientHeight);
    if (isInitialRendering) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js [hunk 2]
```diff
@@ -613,20 +612,162 @@ QUnit.module('MessageList', moduleConfig, () => {
...
-                    assert.roughEqual(scrollTop, this.getScrollOffsetMax(), 1);
+                    assert.notEqual(scrollTop, 0, 'scroll position should change after the element is attached to the DOM');
+                    assert.roughEqual(scrollTop, this.getScrollOffsetMax(), 1, 'scroll position should be at the bottom after attachment');
 
                     done();
                 }, this._resizeTimeout);
             });
         });
+
+        QUnit.test('should save last scroll position after reducing height', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]:
```suggestion
        QUnit.test('should be scrolled to the bottom after reducing height if it's initially scrolled to the bottom', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js [hunk 3]
```diff
@@ -613,20 +612,162 @@ QUnit.module('MessageList', moduleConfig, () => {
...
+                    const scrollTop = this.getScrollable().scrollTop();
+
+                    assert.roughEqual(scrollTop, this.getScrollOffsetMax(), 1, 'max scroll position should be saved after reducing height');
+
+                    done();
+                }, this._resizeTimeout);
+            }, this._resizeTimeout);
+        });
+
+        QUnit.test('should save last scroll position after increasing height', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]:
```suggestion
        QUnit.test('should be scrolled to the bottom after increasing height if it's initially scrolled to the bottom', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js [hunk 4]
```diff
@@ -613,20 +612,162 @@ QUnit.module('MessageList', moduleConfig, () => {
...
+                    const scrollTop = this.getScrollable().scrollTop();
+
+                    assert.roughEqual(scrollTop, this.getScrollOffsetMax(), 1, 'max scroll position should be saved after increasing height');
+
+                    done();
+                }, this._resizeTimeout);
+            }, this._resizeTimeout);
+        });
+
+        QUnit.test('should save last scroll position after reducing height from middle scrolltop', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]:
```suggestion
        QUnit.test('should update visual scroll position after reducing height if it's not scrolled to the bottom (fix viewport bottom point)', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js [hunk 5]
```diff
@@ -613,20 +612,162 @@ QUnit.module('MessageList', moduleConfig, () => {
...
+                    const scrollTop = this.getScrollable().scrollTop();
+
+                    assert.roughEqual(scrollTop, this.getScrollOffsetMax() - 200, 1, 'scroll position should be set correctly after reducing height');
+
+                    done();
+                }, this._resizeTimeout);
+            }, this._resizeTimeout);
+        });
+
+        QUnit.test('should save its scroll position after increasing height', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]:
```suggestion
        QUnit.test('should keep visual scroll position after increasing height if it's not scrolled to the bottom (fix viewport top point)', function(assert) {
```

Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]: https://github.com/DevExpress/DevExtreme/pull/28121/commits/61f3a7a9b35e3c116d48f225161c9108aa2b627e

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js (hunk 4 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]: https://github.com/DevExpress/DevExtreme/pull/28121/commits/61f3a7a9b35e3c116d48f225161c9108aa2b627e

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]: https://github.com/DevExpress/DevExtreme/pull/28121/commits/61f3a7a9b35e3c116d48f225161c9108aa2b627e

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageList.tests.js]: https://github.com/DevExpress/DevExtreme/pull/28121/commits/61f3a7a9b35e3c116d48f225161c9108aa2b627e

File: packages/devextreme/js/__internal/ui/chat/messagelist.ts (hunk 1 above)
Comment (EugeniyKiyashko) [reply packages/devextreme/js/__internal/ui/chat/messagelist.ts]: https://github.com/DevExpress/DevExtreme/pull/28121/commits/61f3a7a9b35e3c116d48f225161c9108aa2b627e

**Review by marker-dao — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**